src/merge-folders.js

```diff
@@ -3,53 +3,55 @@
 const R = require('ramda')
 const path = require('path')
 
-function isParentFolder (parentPath, childPath) {
-  const relative = path.relative(parentPath, childPath)
-  let retValue = false;
-  
-  if (relative) {
-    if (relative.startsWith('..')) {
+const endsWith = R.curry((ending, str) => str.endsWith(ending))
+const startsWith = R.curry((starting, str) => str.startsWith(starting))
+
+// collapses multiple sequential '**/' parts into a single '**/'
+const normalizeGlob = R.replace(/\*\*([\\/])(?:\*\*[\\/])+/g, '**$1')
+
+const getUniqueFolders = R.compose(
+  R.uniq,
+  R.map(R.compose(normalizeGlob, path.normalize, path.dirname)),
+  R.flatten
+)
+
+const isParentFolder = R.curry((parentPath, childPath) => R.compose(
+  R.ifElse(
+    R.complement(R.isEmpty),
+    R.ifElse(
+      startsWith('..'),
       // ** should always be chosen over any specified folders
-      if (parentPath.endsWith('**') && relative.split(path.sep)[1] !== '..') {
-        retValue = true
-      }
-    } else {
-      retValue = true
-    }
-  }
+      R.compose(
+        R.and(endsWith('**')(parentPath)),
+        R.compose(R.not, R.equals('..'), R.nth(1), R.split(path.sep))
```

Comment:
can this be separated into a named variable? I am getting confused what it does just by looking at it